x-pack/test/functional/page_objects/license_management_page.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { KibanaFunctionalTestDefaultProviders } from '../../types/providers';
+
+export const LicenseManagementPageProvider = ({
+  getService,
+}: KibanaFunctionalTestDefaultProviders) => {
+  const testSubjects = getService('testSubjects');
+  const log = getService('log');
+
+  return {
+    async appHasLoaded() {
+      await log.debug('Checking for license header.');
+      const licenseText = await testSubjects.getVisibleText('licenseText');
```

Comment:
I would have called the function `licenseText()` and returned the page object from the testSubject (as the file is a "page_objects" file).

So

```ts
return {
    async licenseText() {
      return await testSubjects.getVisibleText('licenseText');
    },
};
```

And have the `expect` inside the test

```ts
it('Loads the app', async () => {
  await log.debug('Checking for license header.');
  const licenseText = await pageObjects.licenseManagement.licenseText();
  expect(licenseText).to.be('Your Trial license is active');
});
```

What do you think?

Reply:
Typically, in the page object model, you want to reserve methods for actions and properties of the page. Not usually individual elements. However, I do recognize that this is the way that Kibana's automation has been built. Since it doesn't hurt the tests or waste anyones time, I am fine doing it this way.